packages/gatsby/src/utils/webpack.config.js

```diff
@@ -343,6 +343,19 @@ module.exports = async (
         PnpWebpackPlugin,
       ],
     }
+
+    const target =
+      stage === `build-html` || stage === `develop-html` ? `node` : `web`
+    if (target === `web`) {
+      // force to use es modules when importing internals of @reach.router
```

Comment:
Should we open an issue on @reach/router to make any changes?

Reply:
We do import internal `match` function (to match those parameterized / wildcarderded paths like `/app/:someParam` or `/app/*`). If `@reach/router` exported that from entry point, we wouldn't need this part, but I'm not sure if Ryan would really want that - as this is lower level function. Worth asking at least ;)

Reply:
@pieh are you sure we need this? https://github.com/reach/router/blob/master/package.json#L6 should take care of it.